File: JavaScript/8js 1/checkForFactor.js
/*
|--------------------------------------------------------------------------
| Grasshopper - Check for factor
|--------------------------------------------------------------------------
|
*/

function checkForFactor(base, factor) {
   return base % factor == 0;
}


function checkForFactor(base, factor) {
   return !(base % factor);
}


function checkForFactor(base, factor) {
   return (Number.isInteger(base / factor))
}


function checkForFactor(base, factor) {
   return base % factor === [].length
}


console.log(checkForFactor(63, 7), true)
console.log(checkForFactor(2450, 5), true)
console.log(checkForFactor(653, 7), false)
console.log(checkForFactor(2453, 5), false)

File: Python/8py 1/check_for_factor.py
"""
|--------------------------------------------------------------------------
| Grasshopper - Check for factor
|--------------------------------------------------------------------------
|
"""

def check_for_factor(base, factor):
   return base % factor == 0


def check_for_factor(base, factor):
   return not (base % factor)


def check_for_factor(base, factor):
   return (base / factor).is_integer()


def check_for_factor(base, factor):
   return factor in (x for x in range(1, base) if base % x == 0)


print(check_for_factor(63, 7), True)
print(check_for_factor(2450, 5), True)
print(check_for_factor(653, 7), False)
print(check_for_factor(2453, 5), False)